src/main/common/filter.c

```diff
@@ -67,6 +67,32 @@ float pt1FilterApply4(pt1Filter_t *filter, float input, uint8_t f_cut, float dT)
     return filter->state;
 }
 
+// Slew filter with limit
+
+void slewFilterInit(slewFilter_t *filter, float slewLimit, float threshold)
+{
+    filter->state = 0.0f;
+    filter->slewLimit = slewLimit;
+    filter->threshold = threshold;
+}
+
+float slewFilterApply(slewFilter_t *filter, float input)
+{
```

Comment:
filter name is poorly chosen - I would expect it to limit slew rate of signal.

Also, it will unlatch only when output returns within slewLimit. If the (correct) signal jumps from above `threshold` to less than  `filter->state - filter->slewLimit`, it will lock indefinitely (PID will spin quad in opposite direction to compensate, signal will never return)

Reply:
Yes, slewFilter is not really a good name. I couldn't think of anything better.